### DISCUSSION TITLE
Best way to backup or migrate?

### DISCUSSION THREAD

**Discussion by nickian** [uxbox/uxbox#1434] [Q&A]
Is there an easy way to backup the entire Penpot application? For example, what if I wanted to take the entire app and data I currently have running and move it to another server? How would I do this?

Is there also a way to automate backups of all of the projects onto the local file system?

Thanks!

**Comment by ghost** [comment]
Use docker backup: https://help.penpot.app/technical-guide/getting-started/#backup-penpot

This should be the easiest way to move to another server.

For backups i use the following method for backup the data, not the whole container:

Copy volume assets to a local folder `$ docker cp penpot-penpot-backend-1:/opt/data/assets .`

Mount asset volume to local folder:
Edit docker-compose.yaml and set local path.

```
    volumes:
      # - penpot_assets:/opt/data/assets
      # 2 times, backend and frontend
      - ./assets:/opt/data/assets
```

SQL-Backup:
`$ docker exec -i penpot-penpot-postgres-1 /usr/bin/pg_dump -U penpot penpot > penpot-penpot-postgres-1.sql`
or gz
`$ docker exec -i penpot-penpot-postgres-1 /usr/bin/pg_dump -U penpot penpot | gzip > penpot-penpot-postgres-1.sql.gz`

**Reply by mjbartholomew** [reply]
Thanks for sharing this. When it comes time to restore the volumes, was steps do you take for the docker volumes?

**Comment by thefonso** [comment]
Maybe this might help. How to make a backup from inside Penpot. https://help.penpot.app/user-guide/import-export/
